src/browser/BrowserAction.cpp

```diff
@@ -477,50 +517,58 @@ const QByteArray BrowserAction::decrypt(const QString encrypted, const QString n
 
     if (m.size() > 0 && n.size() > 0 && ck.size() > 0 && sk.size() > 0) {
         if (crypto_box_open_easy(d.data(), m.data(), ma.length(), n.data(), ck.data(), sk.data()) == 0) {
-            result = getQByteArray(d.data(), std::char_traits<char>::length(reinterpret_cast<const char *>(d.data())));
+            return getQByteArray(d.data(), std::char_traits<char>::length(reinterpret_cast<const char *>(d.data())));
         }
     }
 
-    return result;
+    return QByteArray();
 }
 
-// Static functions
-////////////////////////////
-
-const QString BrowserAction::getBase64FromKey(const uchar* array, const uint len)
+QString BrowserAction::getBase64FromKey(const uchar* array, const uint len)
 {
     return getQByteArray(array, len).toBase64();
 }
 
-const QByteArray BrowserAction::getQByteArray(const uchar* array, const uint len) const
+QByteArray BrowserAction::getQByteArray(const uchar* array, const uint len) const
 {
     QByteArray qba;
+    qba.reserve(len);
     for (uint i = 0; i < len; ++i) {
         qba.append(static_cast<char>(array[i]));
     }
     return qba;
 }
 
-const QJsonObject BrowserAction::getJSonObject(const uchar* pArray, const uint len) const
+QJsonObject BrowserAction::getJsonObject(const uchar* pArray, const uint len) const
 {
     QByteArray arr = getQByteArray(pArray, len);
     QJsonParseError err;
     QJsonDocument doc(QJsonDocument::fromJson(arr, &err));
     return doc.object();
 }
 
-const QJsonObject BrowserAction::getJSonObject(const QByteArray ba) const
+QJsonObject BrowserAction::getJsonObject(const QByteArray ba) const
 {
     QJsonParseError err;
     QJsonDocument doc(QJsonDocument::fromJson(ba, &err));
     return doc.object();
 }
 
-const QByteArray BrowserAction::base64Decode(const QString str)
+QByteArray BrowserAction::base64Decode(const QString str)
 {
     return QByteArray::fromBase64(str.toUtf8());
 }
 
+QString BrowserAction::incrementNonce(const QString& nonce)
+{
+    return nonce; // Fix this later
```

Comment:
...

Reply:
Really. I'll fix this soon. This is here just because I can test all the changes with the current browser extension release.